dev-infrastructure/scripts/istio.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+echo "********** ISTIO Upgrade Started **************"
+# Followed this guide for istio upgrade https://learn.microsoft.com/en-us/azure/aks/istio-upgrade
+# Get the current istio and check if it match target version
+export CURRENTVERSION=$(kubectl get pods -n aks-istio-system -o jsonpath='{.items[*].metadata.name}' | tr ' ' '\n' | tail -n +2 | head -n 1 | cut -d '-' -f 2,3,4)
```

Comment:
The name of the pod is not a durable way to check for the version. Please look into `istioctl tag set` command - if it is idempotent, we do not need to have this check for versions whatsoever.